src/lib/lockfile.ts

```diff
@@ -0,0 +1,117 @@
+import * as fs from 'fs';
+import { dirname } from 'path';
+import { isRight } from 'fp-ts/lib/Either';
+
+import { exec, unlinkAll } from './fs-utils';
+import { NumericIdentifier } from '../types';
+
+// Equivalent to `drwxrwxrwt`
+const STICKY_WRITE_PERMISSIONS = 0o1777;
+export const BASE_LOCK_DIR =
```

Comment:
I think this constant needs to go on `update-lock.ts` as this module provides generic lock functionality independent from the supervisor.